ci-operator/step-registry/azure/provision/service-principal/hypershift/azure-provision-service-principal-hypershift-commands.sh

```diff
@@ -25,34 +25,35 @@ declare -A component_to_client_id
 declare -A component_to_cert_name
 
 for component in $COMPONENTS; do
-    name="${SP_NAME_PREFIX}-${component}"
-    scopes="/subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_HC"
-    if [[ $component == ingress ]]; then
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_VNET"
-    elif [[ $component == cloud-provider ]]; then
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_NSG"
-    elif [[ $component == cpo ]]; then
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_NSG"
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_VNET"
-    elif [[ $component == capz ]]; then
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_NSG"
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_VNET"
-    fi
+   name="${SP_NAME_PREFIX}-${component}"
+# by default, we should enable HYPERSHIFT_AZURE_ASSIGN_SERVICE_PRINCIPAL_ROLES to assign roles to the service principal automatically
```

Comment:
I think this is missing some components and some coverage. See https://github.com/openshift/hypershift/blob/92bbcedfa4951be1c4edbba6465342a37fb6fda7/cmd/infra/azure/create.go#L789-L788

Reply:
thanks. I compared the original QE permission.
```
	case nodePoolMgmt:
		scopes = append(scopes, vnetRG)
```
we should add nsgRG for the capz